File: src/main/java/com/rs/java/game/area/banks/CatherbyBank.java
package com.rs.java.game.area.banks;

import com.rs.java.game.WorldTile;
import com.rs.java.game.area.Area;
import com.rs.java.game.area.Shape;
import com.rs.java.game.area.shapes.Polygon;

public final class CatherbyBank extends Area {

	@Override
	public Area update() {
		return this;
	}

	@Override
	public String name() {
		return "Catherby Bank";
	}

	public CatherbyBank() {
		super(
				new Polygon(new WorldTile[] {
						new WorldTile(2806, 3438, 0),
						new WorldTile(2806, 3441, 0),
						new WorldTile(2812, 3441, 0),
						new WorldTile(2812, 3438, 0),
						new WorldTile(2806, 3438, 0) // closing point
				})
		);
	}

	@Override
	public boolean member() {
		return false;
	}

	@Override
	public Environment environment() {
		return Environment.SAFEZONE;
	}

}

File: src/main/java/com/rs/kotlin/game/world/area/areas/banks/CatherbyBank.kt
package com.rs.kotlin.game.world.area.areas.banks

import com.rs.java.game.WorldTile
import com.rs.kotlin.game.world.area.Area
import com.rs.kotlin.game.world.area.shape.Polygon

class CatherbyBank : Area(
    Polygon(
        arrayOf(
            WorldTile(2806, 3438, 0),
            WorldTile(2806, 3441, 0),
            WorldTile(2812, 3441, 0),
            WorldTile(2812, 3438, 0),
            WorldTile(2806, 3438, 0) // closing point
        )
    )
) {
    override fun update(): Area {
        return this
    }

    override fun name(): String {
        return "Catherby Bank"
    }

    override fun member(): Boolean {
        return false
    }

    override fun environment(): Environment {
        return Environment.SAFEZONE
    }
}
